subsys/mgmt/mcumgr/smp/src/smp.c

```diff
@@ -191,14 +191,20 @@ static int smp_handle_single_payload(struct smp_streamer *cbuf, const struct smp
 		bool ok;
 
 		*handler_found = true;
-		ok = zcbor_map_start_encode(cbuf->writer->zs,
-					    CONFIG_MCUMGR_SMP_CBOR_MAX_MAIN_MAP_ENTRIES);
+#if defined(CONFIG_MCUMGR_MGMT_CUSTOM_PAYLOAD)
+		if (!handler->custom_payload) {
+#endif
+			ok = zcbor_map_start_encode(cbuf->writer->zs,
+								  CONFIG_MCUMGR_SMP_CBOR_MAX_MAIN_MAP_ENTRIES);
```

Comment:
this one has gone haywire

Reply:
Trying again, it's not easy to make a nice alignment, my vscode shows the alignment completely different.